avaframe/ana4Stats/probAna.py

```diff
@@ -327,3 +429,183 @@ def makeDictFromVars(cfg):
+    if cfgProb['PROBRUN']['sampleMethod'].lower() == 'latin':
+        sampler = qmc.LatinHypercube(d=len(varParList), seed=randomGen)
+        sample = sampler.random(n=int(cfgProb['PROBRUN']['nSample']))
+        sample = qmc.scale(sample, lowerBounds, upperBounds)
+        log.info('Parameter sample created using latin hypercube sampling')
+    else:
+        message = ('Sampling method: %s not a valid option' % cfgProb['PROBRUN']['sampleMethod'])
+        log.error(message)
+        raise AssertionError(message)
+
+    # create dictionary with all the info
+    paramValuesD = {'names': varParList,
+                    'values': sample,
+                    'typeList': cfgProb['PROBRUN']['varParType'].split('|'),
+                    'thReadFromShp': thReadFromShp}
+
+    return paramValuesD
+
+
+def createCfgFiles(paramValuesD, comMod, cfg, cfgPath='', fileOverride=''):
+    """ create all config files required to run com Module from parameter variations using paramValues
+
+        Parameters
+        -----------
+        paramValuesD: dict
+            dictionary with parameter names and values (array of all sets of parameter values, one row per value set)
+        comMod: com module
+            computational module
+        cfg: configparser object
+            configuration settings
+        cfgPath: str
+            path where cfg files should be saved to
+        fileOverride: pathlib path
+            if config of comMod should be overwritten provide file path
+
+        Returns
+        --------
+        cfgFiles: list
+            list of cfg file paths for comMod including the updated values of the parameters to vary
+
+    """
+
+    # get filename of module
+    modName = str(pathlib.Path(comMod.__file__).stem)
+
+    # read initial configuration
+    if fileOverride != '' and cfg['PROBRUN'].getboolean('defaultComModuleCfg'):
+        message = 'fileOverride provided AND defaultComModuleCfg set to True, only one is allowed'
+        log.error(message)
+        raise AssertionError(message)
+    else:
+        cfgStart = cfgUtils.getModuleConfig(comMod, fileOverride=fileOverride, toPrint=False,
+                                            onlyDefault=cfg['PROBRUN'].getboolean('defaultComModuleCfg'))
+
+    # create one cfgFile with one line of the parameter values from the full parameter variation
+    cfgFiles = []
+    for count1, pVal in enumerate(paramValuesD['values']):
+        for index, par in enumerate(paramValuesD['names']):
+            # convert percent variation to a +- variation in percent and of step 1
+            if par in paramValuesD['thReadFromShp']:
+                thVal = pVal[index] - 100.
+                if thVal > 0.:
+                    cfgStart['GENERAL'][par + 'PercentVariation'] = '+' + str(thVal) + '$1'
+                else:
+                    cfgStart['GENERAL'][par + 'PercentVariation'] = str(thVal) + '$1'
+            else:
+                cfgStart['GENERAL'][par] = str(pVal[index])
```

Comment:
This looks convoluted and hard to understand. Would some kind of list-comprehension work here?

Reply:
thVal = pVal[index] - 100.
signVal = ['+' if thVal > 0 else '']
cfgStart['GENERAL'][par + 'PercentVariation'] = signVal + str(thVal) + '$1'

Reply:
Uh, sorry, I was thinking more about lines 593, and 594. But then I looked at the contents of paramValuesD...

Might it be a better solution to have the paramValuesD as a pandas dataframe? Then the manipulation would be a bit easier/more obvious.